sympy/sets/handlers/intersection.py

```diff
@@ -435,10 +435,20 @@ def _(a, b):
             start = a.start
             left_open = a.left_open
         else:
-            #this is to ensure that if Eq(a.start,b.start) but
-            #type(a.start) != type(b.start) the order of a and b
-            #does not matter for the result
-            start = list(ordered([a,b]))[0].start
+            start = a.start
+            if a.start != b.start:
+                # For example Integer(2) != Float(2)
+                # Prefer the Float boundary because Floats should be
+                # contagious in calculations.
+                if b.start.has(Float) and not a.start.has(Float):
```

Comment:
This change is another good candidate to split out to a separate PR.

Reply:
As mentioned in #25662 in which this was placed, this change will not work until this PR is in.